sdk/eventhub/event-hubs/test/eventHubConsumerClient.spec.ts

```diff
@@ -225,6 +225,44 @@ describe("EventHubConsumerClient", () => {
       await producerClient.close();
     });
 
+    describe("#close()", function(): void {
+      it("stops any actively running subscriptions", async function(): Promise<void> {
+        const subscriptions: Subscription[] = [];
+        const client = new EventHubConsumerClient(
+          EventHubConsumerClient.defaultConsumerGroupName,
+          service.connectionString,
+          service.path
+        );
+
+        // Spin up multiple subscriptions.
+        for (const partitionId of partitionIds) {
+          subscriptions.push(
+            client.subscribe(partitionId, {
+              async processError() {
+                /* no-op for test */
+              },
+              async processEvents() {
```

Comment:
Does it matter that they should get a processClose() event? I know it's probably impossible for them not to, but might be nice to see for the test.

Reply:
I was trying to keep this test simple and just test the one thing (that we close the subscriptions). I don't think any of the handlers get called since I close the client so quickly. I can add another test to ensure close gets called though once it starts reading from a partition.

Reply:
I added a test that ensures the customer gets a `Shutdown` reason in `processClose`. Turns out writing this test found an issue with some code paths that were causing #8598, so added a commit to fix those up too.